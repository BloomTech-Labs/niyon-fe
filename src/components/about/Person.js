import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { Link } from "react-router-dom";
import './styles.scss';

const useStyles = makeStyles((theme) => ({
  root: { display: 'flex' },
  person: {
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
    justifyContent: "center",
    textAlign: "center",
    margin: "32px",
    width: "250px"
  },
  personImg: {
    width: "150px",
    height: "150px",
    borderRadius: "50%",
    overflow: "hidden",
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
    boxShadow: "0 5px 25px rgba(0,0,0,0.2)",
    marginBottom: "10px",
  },
  h2: {
    margin: "0",
    marginTop: "10px"
  },
  p: {
    color: "#484848",
    margin: "0",
    marginTop: "10px"
  },
  img2: {
    width: "160px",
    height: "auto"
  }
}));

function Person({src, name, position, link}) {
  const classes = useStyles();
  return (
    <div className={classes.person} data-test="person-container">
      <div className={classes.personImg} data-test="person-image-container">
        <a href={link}><img className={classes.img2} src={src} alt=""/></a>
        {/* // can make the link pop up in a window with 'target="_blank"// */}
      </div>
      <h2>{name}</h2>
      <p>{position}</p>
    </div>
  );
};

export default Person;
