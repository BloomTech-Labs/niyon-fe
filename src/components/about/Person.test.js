import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Person from './Person';
import findByTestAttr from '../../tests/utils';


const setUp = (props={}) => {
   let component = shallow(<Person { ...props } />);
   return component;
}

describe('<Person /> component testing',() => {
   let component;
   const testProps = {
      name: 'test-name',
      position: 'test-position'
   }
   beforeEach(() => {
      component = setUp(testProps);
   });

   it('should pass the snapshot test correctly', () => {
       expect(toJSON(component)).toMatchSnapshot();
   });

   it('should render <Person /> component with no errors', () => {
      expect(component.length).toBe(1);
   });

   it('should render Person container',() => {
      const personContainer = findByTestAttr(component, 'person-container');
      expect(personContainer.length).toBe(1);
   });

   it('should render Person Image container',() => {
      const personImageContainer = findByTestAttr(component, 'person-image-container');
      expect(personImageContainer.length).toBe(1);
   });

   it('should render anchor tag for image link', () => {
      const a = component.find('a');
      expect(a.length).toBe(1);
   });

   it('should render <h2 /> correctly', () => {
       const header = component.find('h2');
       expect(header.exists()).toBe(true);      
   });

   it('should render header content correctly', () => {
      const header = component.find('h2');      
      expect(header.text()).toEqual(testProps.name);
   });

   it('should render <p/> correctly with its contents', () => {
       const paragraph = component.find('p');
       expect(paragraph.length).toBe(1);
       expect(paragraph.text()).toEqual(testProps.position);
   });

});
