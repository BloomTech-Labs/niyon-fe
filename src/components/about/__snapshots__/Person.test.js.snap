// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Person /> component testing should pass the snapshot test correctly 1`] = `
<div
  className="makeStyles-person-2"
  data-test="person-container"
>
  <div
    className="makeStyles-personImg-3"
    data-test="person-image-container"
  >
    <a>
      <img
        alt=""
        className="makeStyles-img2-6"
      />
    </a>
  </div>
  <h2>
    test-name
  </h2>
  <p>
    test-position
  </p>
</div>
`;
